src/current/v25.2/changefeed-message-envelopes.md

```diff
@@ -0,0 +1,795 @@
+
+#### `"ts_ns"` 
+
+{% include_cached new-in.html version="v25.2" %} The processing time of the event by the changefeed job.
+
+#### `table`: 
+
+The name of the table that generated the change. This field appears in certain contexts:
+
+- In sinkless changefeeds (those created with [`EXPERIMENTAL CHANGEFEED FOR`]({% link {{ page.version.version }}/changefeed-for.md %}) or `CREATE CHANGEFEED ... WITH sinkless`), the output includes a `"table"` field for each row change, because a single sinkless feed could cover multiple tables.
+- In the [`enriched`](#enriched) envelope, the [fully qualified table name]({% link {{ page.version.version }}/sql-name-resolution.md %}) is typically part of the [`"source"`](#source) field (as `"database"`, `"schema"`, `"table"` sub-fields), rather than a separate top-level `"table"` field.
+
+{{site.data.alerts.callout_info}}
+When a changefeed targets a table with multiple column families, the family name is appended to the table name as part of the topic. Refer to [Tables with columns families in changefeeds]({% link {{ page.version.version }}/changefeeds-on-tables-with-column-families.md %}#message-format) for guidance.
+{{site.data.alerts.end}}
+
+#### `__crdb__`
+
+A field used by the [`bare`](#bare) envelope to carry CockroachDB metadata that would otherwise be top level. When `envelope=bare`, the message’s top level is the row data, so CockroachDB inserts any needed metadata (primary key, topic, timestamps, etc.) under a nested `"__crdb__"` object. For example:
+
+~~~json
+{
+"__crdb__": {"key": [101]},
+"id": 101,
+"customer_id": 1,
+"total": 50.00,
+"status": "new"
+}
+~~~
+
+Here `"__crdb__": {"key": [101]}` holds the primary key for the row, while the rest of the object are the row’s columns at the top level. Other metadata like [`"updated"`](#updated) timestamps or [`resolved`](#resolved-option) timestamps would also appear inside `__crdb__`, if present. This field is specific to the `bare` envelope or other cases (like custom [CDC queries]({% link {{ page.version.version }}/cdc-queries.md %}) with `SELECT`) where metadata needs to be attached without interfering with the selected row data.
+
+### `"schema"` 
+
+{% include_cached new-in.html version="v25.2" %} The schema and type of each payload field. This is included when using [`envelope=enriched`](#enriched-option) with [`enriched_properties='schema'`](#enriched-properties-option). The `"schema"` field provides information needed to interpret the data. The following are present for each schema present in the envelope, depending on the configured options:
+
+- `"field"`: The name of field.
+- `"type"`: The type of the field.
+- `"optional"`: This is a boolean field that defines whether a field may be absent in the `"payload"` section of the envelope depending on the configuration.
+- `"name"`: The name of the described schema.
```

Comment:
if type=array then `items` will have the array element type

Reply:
I believe this is now covered.